src/QMCDrivers/Optimizers/DescentEngine.h

```diff
@@ -130,19 +144,27 @@ class DescentEngine
   ////////////////////////////////////////////////////////////////////////////////////////////////////////
   void sample_finish();
 
-  const std::vector<double>& getAveragedDerivatives() { return LDerivs; }
+  const std::vector<double>& getAveragedDerivatives() const { return LDerivs; }
 
   // helper method for updating parameter values with descent
   void updateParameters(int stepNum, int descentNum);
 
   //helper method for seting step sizes for different parameter types in descent optimization
   double setStepSize(int i);
 
+  //stores derivatives so they can be used in accelerated descent algorithm on later iterations
   void storeDerivRecord() { derivRecords.push_back(LDerivs); }
 
+  //helper method for transferring information on parameter names and types to the engine
   void setupUpdate(const optimize::VariableSet& myVars);
 
-  std::vector<double> retrieveNewParams() { return currentParams; }
+  void storeVectors(std::vector<double>& currentParams);
+
+  const int retrieveStoreFrequency() const { return store_num; }
+
+  const std::vector<std::vector<double>> retrieveHybridBLM_Input() const { return hybridBLM_Input; }
```

Comment:
Here return const reference instead of value
`const std::vector<std::vector<double>>&`